[deconz] StringIndexOutOfBoundsException with short Bridge name

Hi everyone. I have the problem that the openhab did not receive any statuses from deconz
I realized that there was the message: Destroyed container cannot be restarted
So I changed the port to 8080 as described in the openhab community.

I think I made the error to delete the API key and retrieve a new one.
Now openhab was not able to retreive a new key anymore (EofException)
```

Caused by: org.eclipse.jetty.io.EofException
	at org.eclipse.jetty.io.ChannelEndPoint.flush(ChannelEndPoint.java:283) ~[?:?]
	at org.eclipse.jetty.io.WriteFlusher.flush(WriteFlusher.java:422) ~[?:?]
	at org.eclipse.jetty.io.WriteFlusher.write(WriteFlusher.java:277) ~[?:?]
	at org.eclipse.jetty.io.AbstractEndPoint.write(AbstractEndPoint.java:381) ~[?:?]
	at org.eclipse.jetty.client.http.HttpSenderOverHTTP$HeadersCallback.process(HttpSenderOverHTTP.java:249) ~[?:?]
	at org.eclipse.jetty.util.IteratingCallback.processing(IteratingCallback.java:241) ~[?:?]
	at org.eclipse.jetty.util.IteratingCallback.iterate(IteratingCallback.java:223) ~[?:?]
	at org.eclipse.jetty.client.http.HttpSenderOverHTTP.sendHeaders(HttpSenderOverHTTP.java:62) ~[?:?]
	at org.eclipse.jetty.client.HttpSender.send(HttpSender.java:212) ~[?:?]
	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.send(HttpChannelOverHTTP.java:85) ~[?:?]
	at org.eclipse.jetty.client.HttpChannel.send(HttpChannel.java:128) ~[?:?]
	at org.eclipse.jetty.client.HttpConnection.send(HttpConnection.java:201) ~[?:?]
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP$Delegate.send(HttpConnectionOverHTTP.java:252) ~[?:?]
	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.send(HttpConnectionOverHTTP.java:121) ~[?:?]
	at org.eclipse.jetty.client.http.HttpDestinationOverHTTP.send(HttpDestinationOverHTTP.java:38) ~[?:?]
	at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:346) ~[?:?]
	at org.eclipse.jetty.client.HttpDestination.process(HttpDestination.java:304) ~[?:?]
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:294) ~[?:?]
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:269) ~[?:?]
	at org.eclipse.jetty.client.HttpDestination.send(HttpDestination.java:246) ~[?:?]
	at org.eclipse.jetty.client.HttpClient.send(HttpClient.java:580) ~[?:?]
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:728) ~[?:?]
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:720) ~[?:?]
	at org.openhab.binding.deconz.internal.netutils.AsyncHttpClient.doNetwork(AsyncHttpClient.java:103) ~[?:?]
	at org.openhab.binding.deconz.internal.netutils.AsyncHttpClient.post(AsyncHttpClient.java:52) ~[?:?]
	at org.openhab.binding.deconz.internal.handler.DeconzBridgeHandler.requestApiKey(DeconzBridgeHandler.java:248) ~[?:?]
	at org.openhab.binding.deconz.internal.handler.DeconzBridgeHandler.lambda$0(DeconzBridgeHandler.java:127) ~[?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_252]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_252]


```
Then I changed the name of the bridge to a very short one and now I receive a StringIndexOutOfBoundsException

2020-11-13 16:34:20.362 [ERROR] [ome.core.thing.internal.ThingManagerImpl] - Exception occurred while calling thing handler ```
factory 'org.openhab.binding.deconz.internal.DeconzHandlerFactory@25f4988e': String index out of range: -4

java.lang.StringIndexOutOfBoundsException: String index out of range: -4

	at java.lang.String.substring(String.java:1927) ~[?:1.8.0_252]

	at org.openhab.binding.deconz.internal.handler.DeconzBridgeHandler.<init>(DeconzBridgeHandler.java:86) ~[?:?]

	at org.openhab.binding.deconz.internal.DeconzHandlerFactory.createHandler(DeconzHandlerFactory.java:88) ~[?:?]

	
```

I wrote the whole story so maybe the EofException is related to the one with StringIndexOutOfBounds.
Is there a problem with too short names? Normally it should not, in my opinion

If Eof is not related should I open a new issue?

--- a/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java
+++ b/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/DeconzBridgeHandler.java
@@ -83,7 +83,11 @@ public class DeconzBridgeHandler extends BaseBridgeHandler implements WebSocketC
         this.http = http;
         this.gson = gson;
         String websocketID = thing.getUID().getAsString().replace(':', '-');
-        websocketID = websocketID.length() < 3 ? websocketID : websocketID.substring(websocketID.length() - 20);
+        if (websocketID.length() < 4) {
+            websocketID = "openHAB-deconz-" + websocketID;
+        } else if (websocketID.length() > 20) {
+            websocketID = websocketID.substring(websocketID.length() - 20);
+        }
         this.websocket = new WebSocketConnection(this, webSocketFactory.createWebSocketClient(websocketID), gson);
     }
 

